packages/substrate-client/src/archive/archive.ts

```diff
@@ -0,0 +1,78 @@
+import { abortablePromiseFn } from "@/internal-utils"
+import { type ClientRequest } from "../client"
+import { createStorageCb } from "./storage-subscription"
+import { createStorageFn } from "./storage"
+import { Archive } from "./public-types"
+import { CallError, InvalidBlockHashError } from "./errors"
+
+const identity =
+  <T>() =>
+  (x: T): T =>
+    x
+
+const handleInvalidBlockHash =
+  <T>() =>
+  (result: T | null, hash: string): T => {
+    if (result === null) throw new InvalidBlockHashError(hash)
```

Comment:
The spec says it can give back `null` result if the block is not found. I'd change the name of this error to have it closer to this meaning... Maybe `BlockHashNotFoundError`?

Reply:
yeah, good point!